Dodge.js

```diff
@@ -1,178 +1,201 @@
 import { StatusBar } from 'expo-status-bar';
+import React, { Component, PureComponent } from 'react';
 import { StyleSheet, Text, View, Dimensions, Button, Image } from 'react-native';
-import { Knight, Circle, Brick, Grenade, MoveKnight, Shrapnel, counter} from "./Entities.js";
+import {
+  Knight, Circle, Brick, Grenade, MoveKnight,
+  Shrapnel, Water
+} from "./Entities.js";
 import { GameEngine } from "react-native-game-engine";
+import Matter from "matter-js";
 import AxisPad from 'react-native-axis-pad';
 ////this looks interesting for the attacker placing/throwing projectiles:
 //https://reactnative.dev/docs/panresponder 
 //https://medium.com/yapsody-engineering/hooks-and-function-components-in-react-native-d4e667c90cda
-var ents = {}
+let score = 0
```

Comment:
Global variables are bad :P 
As suggested elsewhere, pass each minigame a reference to the battle screen, and pass the score back through a function there.